Salesforce/salesforce/connector.py

```diff
@@ -0,0 +1,178 @@
+"""Contains connector, configuration and module."""
+
+import asyncio
+from datetime import datetime, timedelta, timezone
+from typing import Any, Optional
+
+import aiocsv
+import aiofiles
+from aiofiles import os as aiofiles_os
+import orjson
+from aiolimiter import AsyncLimiter
+from dateutil.parser import isoparse
+from loguru import logger
+from pydantic import HttpUrl
+from sekoia_automation.connector import Connector, DefaultConnectorConfiguration
+from sekoia_automation.module import Module
+from sekoia_automation.storage import PersistentJSON
+
+from client.http_client import SalesforceHttpClient
+
+
+class SalesforceConfig(DefaultConnectorConfiguration):
+    """Configuration for SalesforceConnector."""
+
+    client_id: str
+    client_secret: str
+    base_url: HttpUrl
+    ratelimit_per_minute: int = 60
+    temp_files_dir: str = "/tmp"
+
+
+class SalesforceModule(Module):
+    """SalesforceConfig."""
+
+    configuration: SalesforceConfig
+
+
+class SalesforceConnector(Connector):
+    """SalesforceConnector class to work with salesforce events."""
+
+    name = "SalesforceConnector"
+
+    module: SalesforceModule
+
+    _salesforce_client: SalesforceHttpClient | None = None
+
+    def __init__(self, *args: Any, **kwargs: Optional[Any]) -> None:
+        """Init SalesforceConnector."""
+
+        super().__init__(*args, **kwargs)
+        self.context = PersistentJSON("context.json", self._data_path)
+        self.from_date = self.last_event_date
+
+    @property
+    def last_event_date(self) -> datetime:
+        """
+        Get last event date.
+
+        Returns:
+            datetime:
+        """
+        now = datetime.now(timezone.utc)
+        one_week_ago = now - timedelta(days=7)
+
+        with self.context as cache:
+            last_event_date_str = cache.get("last_event_date")
+
+            # if undefined, retrieve events from the last 7 days
+            if last_event_date_str is None:
+                return one_week_ago
```

Comment:
I think that getting the events from the last hour, if no last_event_date exists, is preferable. I worry that customers, setting this integration, doesn't understand why they are not viewing their most recent events, in the interface, when starting the connector.